src/OpenTelemetry/Metrics/Processors/MeasurementProcessor.cs

```diff
@@ -14,11 +14,15 @@
 // limitations under the License.
 // </copyright>
 
-#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
 
 namespace OpenTelemetry.Metrics
 {
     public abstract class MeasurementProcessor : BaseProcessor<MeasurementItem>
     {
+        public abstract void OnEnd<T>(MeasurementItem measurementItem, ref DateTimeOffset dt, ref T value, ref ReadOnlySpan<KeyValuePair<string, object>> tags)
```

Comment:
Not sure what benefit we gain by extending BaseProcessor here, as this seems to be defining a new OnEnd method.

Could we modify MeasurementItem to have all the things, and then MeasurementProcessor can be simply extending BaseProcessor<MeasurementItem>

(not blocking. just noting some observations in the PR :) )

Reply:
The reason is that I would have to "allocate" memory if I stuffed it into Measurement object. I did try the approach and it kinda work (you can see history of it in commits) but was messy.

Reply:
okay. we may then remove this class being a subclass of BaseProcessor then.

Reply:
I don't understand your comment.

If we removed MeasurementProcessor, we would then have to relocate the extended OnEnd() to each of the other class (currently, AggregateProcessor and TagEnrichmentProcessor) or have these classes inherit a new interface. I need this interface as I need to call it for the hot path update function.

Reply:
I can see your rational for the MetricProcessor class as it's currently empty.